internal/transform/cleanup.go

```diff
@@ -0,0 +1,89 @@
+package transform
+
+import (
+	"context"
+	"github.com/pb33f/libopenapi"
+	v3 "github.com/pb33f/libopenapi/datamodel/high/v3"
+	"github.com/pb33f/libopenapi/orderedmap"
+	"github.com/speakeasy-api/speakeasy-core/openapi"
+	"github.com/speakeasy-api/speakeasy/internal/log"
+	"gopkg.in/yaml.v3"
+	"io"
+	"strings"
+	"unicode"
+)
+
+func CleanupDocument(ctx context.Context, schemaPath string, yamlOut bool, w io.Writer) error {
+	return transformer[interface{}]{
+		schemaPath:  schemaPath,
+		transformFn: Cleanup,
+		w:           w,
+		jsonOut:     !yamlOut,
+	}.Do(ctx)
+}
+
+func Cleanup(ctx context.Context, doc libopenapi.Document, model *libopenapi.DocumentModel[v3.Document], _ interface{}) (libopenapi.Document, *libopenapi.DocumentModel[v3.Document], error) {
+	pathItems := model.Model.Paths.PathItems
+	var pathsToDelete []string
+
+	for pathPair := orderedmap.First(pathItems); pathPair != nil; pathPair = pathPair.Next() {
+		path := pathPair.Key()
+		pathVal := pathPair.Value()
+		operations := pathVal.GetOperations()
+		if operations.Len() == 0 {
```

Comment:
There are paths without operations?

Reply:
Primarily after we do `transform remove-unused`. Unlikely we'd ever see this from a customer spec